src/diffusers/utils/hub_utils.py

```diff
@@ -369,9 +378,13 @@ def _get_model_file(
                 f"'https://huggingface.co/{pretrained_model_name_or_path}' for available revisions."
             )
         except EntryNotFoundError:
-            raise EnvironmentError(
-                f"{pretrained_model_name_or_path} does not appear to have a file named {weights_name}."
-            )
+            # This should correspond to a shard index file.
```

Comment:
why do we need to return something different when we can't find the shard index file?

why can't we just do 
```python
try:
  model_file = _get_model_file(...)
   ...
except ...
  model_file = None
```

Reply:
I guess this question I still have: why do we need to return `None` when we can't find a shard index file? vs for any other file we get find we raise errors - 
where in the code is this needed?